### PR TITLE
Merge `development` into `main` branch

### PR DESCRIPTION
**Pull request by o-rashad**
#### What's new :question: 
1. Learned how to connect Rails with React using `react-rails` gem
2. Get information to flow between the front-end and the back-end using React `dispatch` and Rails Api Controller
3.  Update README file
4. Use `simplecss` framework to apply styles

### CODE REVIEW

**Review by andersonlebon — COMMENTED** [review]
## Hi @od-c0d3r 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that Has linter errors

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by o-rashad** [comment]
Thanks @andersonlebon for the feedback,
The linters were working locally fine, i had to resolve the GitHub Workflow errors.
Now everything is up :+1:

File: app/javascript/components/Greeting.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  useEffect(() => dispatch(fetchMessage()), [])
+
+  return (
+    <React.Fragment>
+      {message.message_content}
+    </React.Fragment>
+  );
+}
+
+export default Greeting
```
Comment (Emmyn5600) [inline app/javascript/components/Greeting.js]: - [x] From the greeting file, You had to add the actions of both `action creator and types` as well as to connect to the rail app to retrieve the random greetings. For example:


 function getGreetings() {
  return dispatch => {
    dispatch({ type: GET_GREETINGS_REQUEST });
    return fetch(`api/grettings.json`)
      .then(response => response.json())
      .then(json => dispatch(getGreetingsSuccess(json)))
      .catch(error => console.log(error));
  };
}; 

export function getGreetingsSuccess(json) {
  return {
    type: GET_GREETINGS_SUCCESS,
    json
  };
};


I see you have implemented the same logic below however try to make it working as expected.

File: config/routes.rb [hunk 2]
```diff
@@ -1,6 +1,13 @@
...
+
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  namespace :api, defaults: { format: 'json' } do
+    get 'greeting', to: 'api#index'
+  end
+
+  get '*path', to: 'static#index', constraints: lambda { |req|
+    !req.xhr? && req.format.html?
+  }
```
Comment (Emmyn5600) [inline config/routes.rb:10]: - [x] None of the random messages is being displayed on the screen when the server is running, this is due to the `No routes matched location "/" `.

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @od-c0d3r 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight :champagne:

- All linters are passing :heavy_check_mark:
- Good commit history :heavy_check_mark:
- Good PR description :heavy_check_mark:

## Required Changes :recycle:

- [ ] Check the comment under the review

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!:clap::clap::clap:

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by od-c0d3r** [comment]
Thanks @Emmyn5600 for your review,
I'm sorry, i wasn't clear on my PR description,
The project requirements didn't specify what is the name the end-point name we should use

could you please try to type `localhost:3000/api/greeting`
and see if it's still not working ?

Thanks

File: config/routes.rb (hunk 2 above)
Comment (o-rashad) [reply config/routes.rb:10]: Thanks @Emmyn5600 for your review,
I'm sorry, i wasn't clear on my PR description,
The project requirements didn't specify what is the name the end-point name we should use

could you please try to type `localhost:3000/api/greeting`
and see if it's still not working ?

Thanks

**Comment by Emmyn5600** [comment]
@od-c0d3r , you are supposed to add the button so that when it is clickable the random message from the sample data in the db/seed has to be displayed. Thanks

File: config/routes.rb (hunk 2 above)
Comment (Emmyn5600) [reply config/routes.rb:10]: @od-c0d3r when I type the localhost:3000/api/greetings I can see the json response for the greetings messages however we need to add the button and when it is clicked the random message has to be displayed.

Comment (Emmyn5600) [reply config/routes.rb:10]: @od-c0d3r when I type the `localhost:3000/api/greetings` I can see the JSON response for the greetings messages however we need to add the button and when it is clicked the random message has to be displayed.

**Comment by o-rashad** [comment]
> @od-c0d3r when I type the localhost:3000/api/greetings I can see the json response for the greetings messages however we need to add the button and when it is clicked the random message has to be displayed.

Thanks again for your kind reply, I think I'm missing something out, the app is working fine on my end.

`localhost:3000/api/greeting` is an API end-point it should return json response.
Any other route Rails will route it to `static#index` controller action which will render  `App` component.

Inside `App` component there are React routes
one of them is `/greeting` which will render `Greeting` component that fetches the created API end-point 
and shows the message

please refer in the project requirements where we should create a button, i can't find it!

would you try this after `rails s`, open a new terminal and try `./bin/webpack-dev-server` ?

Thanks again,

**Comment by Emmyn5600** [comment]
@od-c0d3r, I understand what you are saying, however,  the best approach to display the random message is to create a button called `RandomGreeting`  or any name, and when it is clicked the random message should be displayed on the screen.

**Comment by o-rashad** [comment]
@Emmyn5600 Thanks, 
yep it make sense to have a button to click on ^^!
Working on it :+1:

**Review by Emmyn5600 — APPROVED** [review]
Hi @od-c0d3r 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! :tada: :clap: :clap: :100:

nodding-yes

## Optional suggestions

- [ ] N/A

#### _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!:clap::clap::clap:
#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
------
#### _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**